src/debugger/debug_workflow.ts

```diff
@@ -121,29 +120,38 @@ export class WorkflowContextDebug extends DBOSContextImpl implements WorkflowCon
       },
       this.span
     );
-    let check: RecordedResult<R>;
+    let check: RecordedResult<R> | Error;
 
     const wrappedTransaction = async (client: UserDatabaseClient): Promise<R> => {
       // Original result must exist during replay.
       const tCtxt = new TransactionContextImpl(this.#dbosExec.userDatabase.getName(), client, this, span, this.#dbosExec.logger, funcID, txn.name);
       check = await this.checkExecution<R>(client, funcID);
 
+      if (check instanceof Error) {
+        this.logger.warn(`original transaction ${txn.name} failed with error: ${check.message}`);
+      }
+
       // Execute the user's transaction.
       const result = await txn(tCtxt, ...args);
       return result;
     };
 
     const result = await this.#dbosExec.userDatabase.transaction(wrappedTransaction, txnInfo.config);
+    check = check!;
+
+    if (check instanceof Error) {
+      throw check;
+    }
 
     // If returned nothing and the recorded value is also null/undefined, we just return it
-    if (result === undefined && !check!.output) {
+    if (result === undefined && check.output) {
```

Comment:
```suggestion
    if (result === undefined && !check.output) {
```
Both values need to be null/undefined